### DISCUSSION TITLE
Malayalam Script Shaping

### DISCUSSION THREAD

**Discussion by Chris36676** [harfbuzz/harfbuzz#2807] [Q&A] [answered]
Hi

I am experiencing incorrect shaping when using a certain font with a Malayalam script. 
The font in question is KairaliMal057Uni (KIMAIUI57.otf) that I attach, along with an example Malayalam script. 
When I use this font under MS Windows where Windows takes care of shaping the rendered output is correct. I confirm this by using MS word or a custom app that makes use of a WPF rich text control.  However we have a custom control where Halfbuzz is used for shaping. When I use this font with Harfbuzz some of the glyths are ordered incorrectly.  

I attach a .png file here (screenshots.png) that shows the problem I am having.

Using a different font NotoSansMalayalam Harfbuzz copes fine and the shaped output is correct.

Now my knowledge of complex language typography is very limited. So I was hoping somebody could give me an idea of what may be causing this. Is this a bug with Harfbuzz or a bug with the font in question?

Why would Harfbuzz work fine with the NotoSansMalayalam font but give incorrect results with the KairaliMal057Uni font?

[MalayFontIssue.zip](https://github.com/harfbuzz/harfbuzz/files/5780892/MalayFontIssue.zip)

Thanks
Chris

**Comment by behdad** [comment]
Thanks. Typically we use issues for such reports. I know there's a way to turn this into an issue, but can't find it. Can't tell without debugging what's wrong. But I'm guessing that we are dealing with 

Here's the two clusters in question:
പ്ര
ക്ഷേ

The first one is simply the prebase-reordering Ra.

My first guess was that the font is an old-spec one. But upon inspection, it does have both `mlym` and `mlm2` scripts in GSUB. However, the GSUB has no `pref` feature whatsoever. So that seems to be a case that Uniscribe seems to do something, but we don't.

**Reply by behdad** [reply]
> I know there's a way to turn this into an issue, but can't find it.

Apparently not: https://github.com/github/feedback/discussions/2861

**Comment by behdad** [comment]
![ScreenShots](https://user-images.githubusercontent.com/142891/104233330-8c945380-5406-11eb-97dc-53da6ec01576.png)

**Comment by behdad** [comment]
@jfkthame any ideas?

**Comment by behdad** [comment, marked as answer]
This is being tracked in https://github.com/harfbuzz/harfbuzz/issues/2814
